### PR TITLE
feat: add proquint

### PR DESCRIPTION
**Pull request by Stebalien**
We're removing proquint support from IPNS and making it a multibase. I've chosen `p` or "pro-" so proquints are always fully pronounceable.

### CODE REVIEW

File: rfcs/PRO-QUINT.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# PRO-QUINT
+
+See: https://arxiv.org/html/0901.4016 ([/ipfs/bafybeib5jsyi5igjwhi7hzkfebpvnq2ykbwpxeaaxlkyfyxqvcecoao4qa](https://bafybeib5jsyi5igjwhi7hzkfebpvnq2ykbwpxeaaxlkyfyxqvcecoao4qa.ipfs.dweb.link)).
+
+While the multibase prefix is `q`, the "full" prefix is actually `pro-`. This way, proquints are always easily pronouncable. For example
```
Comment (ribasushi) [inline rfcs/PRO-QUINT.md:5]: Typo 
```suggestion
While the multibase prefix is `p`, the "full" prefix is actually `pro-`. This way, proquints are always easily pronouncable. For example
```

Comment (Stebalien) [reply rfcs/PRO-QUINT.md]: (oops, thanks)

File: rfcs/PRO-QUINT.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# PRO-QUINT
+
+See: https://arxiv.org/html/0901.4016 ([/ipfs/bafybeib5jsyi5igjwhi7hzkfebpvnq2ykbwpxeaaxlkyfyxqvcecoao4qa](https://bafybeib5jsyi5igjwhi7hzkfebpvnq2ykbwpxeaaxlkyfyxqvcecoao4qa.ipfs.dweb.link)).
```
Comment (lidel) [inline rfcs/PRO-QUINT.md:3]: nit: looks better when we use path router in specs:
```suggestion
See: https://arxiv.org/html/0901.4016 ([/ipfs/bafybeib5jsyi5igjwhi7hzkfebpvnq2ykbwpxeaaxlkyfyxqvcecoao4qa](https://dweb.link/ipfs/bafybeib5jsyi5igjwhi7hzkfebpvnq2ykbwpxeaaxlkyfyxqvcecoao4qa)).
```

**Review by lidel — APPROVED** [review]
Merging this as `draft` (to reserve `p`) and seeing if anyone asks us to implement it sounds like a good plan. 

ps. indedd, `pro-` is way better for "pronouncable" identifiers than... `0q-`   
(sic! – thats the "magic numbers" prefix [suggested in the original paper](https://dweb.link/ipfs/bafybeib5jsyi5igjwhi7hzkfebpvnq2ykbwpxeaaxlkyfyxqvcecoao4qa#_conclusion_and_specification) :upside_down_face: )

**Comment by Stebalien** [comment]
> ps. indedd, pro- is way better for "pronouncable" identifiers than... 0q-
(sic! – thats the "magic numbers" prefix suggested in the original paper upside_down_face )

Also, that starts with `0` which is already reserved.

**Pull request merged**